katsdpsigproc/rfi/host.py

```diff
@@ -186,27 +213,29 @@ def __call__(self, deviations, noise):
 class FlaggerHost:
     """Combine host background and thresholding implementations to make a flagger."""
 
-    def __init__(self, background, noise_est, threshold):
+    def __init__(self, background: AbstractBackgroundHost, noise_est: AbstractNoiseEstHost,
+                 threshold: AbstractThresholdHost):
         self.background = background
         self.noise_est = noise_est
         self.threshold = threshold
 
-    def __call__(self, vis, input_flags=None):
+    def __call__(self, vis: np.ndarray, input_flags: Optional[np.ndarray] = None) -> np.ndarray:
         """Perform the flagging.
 
         Parameters
         ----------
-        vis : array-like
```

Comment:
Is array-like equivalent to `np.ndarray`?

Reply:
In numpy docs it's used to mean roughly "anything that can be passed to `np.array`". I had vaguely the same idea, but in general I'm not actually passing it into `np.array` and it won't work unless it is an actual numpy array (or at least something providing the same interface), so I've just been annotating as `np.ndarray`. For now this just ends up being equivalent to `Any` because numpy doesn't have type annotations; it might need revisiting if numpy-stubs makes progress one day.